tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx

```diff
@@ -38,7 +38,61 @@ class RNTupleWriter;
 class RNTupleWriteOptions;
 
 namespace Internal {
-class RPageSinkBuf;
+class RProjectedFields;
+
+RFieldZero &GetFieldZero(RNTupleModel &model);
```

Comment:
This function if used improperly is an invariant/systemic breaking use.   I would recommend making it even more hidden by declaring it into a private header (a header in `/src`)

Reply:
This is not possible because it must be declared `friend` in the `RNTupleModel` class declaration. Using `Internal` functions improperly is something we cannot prevent.

Reply:
Actually you can declare a friend function that is not declared (and with modern C++, this friend declaration is not a declaration).  See https://godbolt.org/z/q6e1EPe9f .